guessing_game.html

```diff
@@ -1,122 +1,269 @@
-<script type="text/javascript">
-  //Constructs a User object with properties that store a chosen date, current date, and number of full moons in between.
-  function User () {
-    //Prompts the user to enter a year, checks it is correctly formatted and not in the future.
-    this.getUserYear = function () {
-      var userYear;
-      userYear =  "";
-      while (userYear.length != 4 || isNaN(userYear)) {
-        userYear = prompt("What year do you choose? (ex: 1989)");
-      }
-      return userYear;
-    };
-
-    // Prompts the user to enter a month, checks it is correctly formatted and a possible month.
-    this.getUserMonth = function () {
-      var userMonth;
-      userMonth = "";
-      while (userMonth.length < 1 || userMonth.length > 2 || isNaN(userMonth) || userMonth > 11) {
-        userMonth = (prompt("What number month do you choose? (ex: 10)") - 1);
-      }
-      return userMonth;
-    };
-
-    //Prompts the user to enter a date, checks it is correctly formatted and a possible date.
-    this.getUserDate = function () {
-      var userDate;
-      userDate = "";
-      while (userDate.length < 1 || userDate.length > 2 || isNaN(userDate) || userDate > 31) {
-        userDate = prompt("What day of the month do you choose? (ex: 29)");
-      }
-      return userDate;
-    };
-
-    //Compares fullUserDate with currentDate, returns true if fullUserDate is in the past, false if it is either the current date or the future.
-    this.compareDates = function () {
-      if ((Date.parse(this.currentDate) - Date.parse(this.fullUserDate)) < 0) {
-        return false;
-      } else {
-        return true;
-      }
-    };
-
-    //Prompts the user to select a date, and checks to ensure that the date is in the past. Declares a currentDate Date() object after fullUserDate to ensure compareDates runs properly.
-    this.getFullUserDate = function() {
-      this.fullUserDate = new Date();
-      this.currentDate = new Date();
-      //Keeps asking for a year until the user selects a year in the past.
-      do {
-        this.fullUserDate.setFullYear(this.getUserYear());
-        if (!this.compareDates()) {
-          alert("You have to choose a year in the past. Try again.");
+<!doctype html>
+<html>
+  <head>
+    <script type="text/javascript">
```

Comment:
Go ahead and move your script tags down to the bottom of the Body, right before the closing body tag. This could cause some issues down the road, with inline JS, where your scripts are running before the DOM tree has been loaded on the page (throwing errors).